xdsl/backend/csl/print_csl.py

```diff
@@ -42,6 +50,45 @@ def print(self, text: str, prefix: str = "", end: str = "\n"):
         for l in text.split("\n"):
             print(self._prefix + prefix + l, file=self.output, end=end)
 
+    @contextmanager
+    def _in_block(self, block_name: str):
+        self.print(f"{block_name} {{")
+        old_prefix = self._prefix
+        self._prefix += self._INDENT
+        yield
+        self._prefix = old_prefix
+        self.print("}")
+        pass
```

Comment:
Does this replace the previous `descend` method?

Reply:
Not really, this just wraps a set of prints in `<block_name> { ... }`. I could combine it with descend, but it's more for cases where we don't have an MLIR `Block` and need to wrap something in braces.

Reply:
I see it only being used for comptime regions, if that is the case, maybe don't need a form this generic? Just some food for thought.

Reply:
It's also used for `layout`, though I can rewrite that as `descend`, since layout does have an MLIR `Block`, then make `_in_block` `_in_comptime` instead, since there are no other CSL block types.

Reply:
Technically, descend can be used to replace all uses of `_in_block`, but this way felt cleaner, since it's always the same pattern:
```
<block-name> {
    <statements>
}
```

Reply:
What would your thoughts be on:

```python
with self._in_block("comptime") as in_comptime:
    in_comptime.print(f"@export_symbol({export_val}, {q_name});")
```

Then we can reuse `descend` in `_in_block` and avoid repetition there?

Reply:
Maybe we can instead replace descend with this instead then? I also feel that the context manager "feels" nicer, I would just argue that we should focus on one method for now.